overlord/devicestate/devicestate.go

```diff
@@ -326,6 +326,21 @@ func CanManageRefreshes(st *state.State) bool {
 	return false
 }
 
+// ResetSession clears the device store session if any.
+func ResetSession(st *state.State) error {
+	device, err := internal.Device(st)
+	if err != nil {
+		return err
+	}
+	if device.SessionMacaroon != "" {
+		device.SessionMacaroon = ""
+		if err := internal.SetDevice(st, device); err != nil {
+			return err
+		}
```

Comment:
Since `device` is a pointer, modifying the session macharoon would change the device structure inside `internal`; should we maybe restore the macaroon if `internal.SetDevice` fails, or operate on a copy?

Unless I'm wrong, this is not a problem currently, because internal.Device always returns a new struct, but if its implementation was changed to implement some caching, then it would become a problem?

Reply:
because the state is actually json data, we assume everywhere that accessors make copies

Reply:
OK!